### ISSUE TITLE
[chm-v1-systems] 100M matched-LM L4 training preflight — non-scientific

### ISSUE THREAD

**Issue by vinceackermann2-sys** [vinceackermann2-sys/tam-research#983]
## Classification

`CHM_V1_100M_STAGE_B_L4_SYSTEMS_PREFLIGHT_ONLY`

This is the separately preregistered Stage B required by #977 after Stage A merged. It is **systems engineering only**. It does not authorize the Stage C scientific scaling screen, does not consume reserved scientific seed `977001`, and does not reopen #959/#973.

## Starting authority

Re-audited immediately before this issue was created:
- live `main`: `118a743f7eb1c224af71bd97dac0f53b2e565d73`
- live tree: `a69e7698f754fb943c9f189d0623e2a6b02daeb6`
- merged Stage A PR: #979
- Stage A exact models: LOCAL `101,803,520` params; EIEM-FLAT `101,836,800` params; mismatch ~`0.0327%`
- frozen future scientific screen budget: `33,554,432` tokens/model
- reserved future scientific seed `977001` remains unauthorized

## Fresh engineering seed

Exactly one non-scientific Stage-B seed is frozen:
- `977201`

It may be used once for this Stage-B GPU preflight only. It has no scientific credit. Never substitute or reuse `19591`, `19592`, `19593`, `971001`, `971002`, `973001`, `973002`, `977001`, or Stage-A smoke seed `977099`.

## Frozen hardware / paid envelope

Provider: Modal
- GPU: exactly `1x NVIDIA L4`
- CPU: 4 physical cores
- RAM: 16 GiB
- GPU function timeout: 1200 s
- retries: 0
- maximum billed compute for the one Stage-B attempt: `$0.50`
- trigger/result namespace must be unique and durable before allocation

The trigger workflow must re-read live Modal rates immediately before allocation and fail closed if the 1200-second worst-case L4+CPU+RAM resource bound exceeds `$0.50`.

## Frozen training geometry

Use the exact Stage-A model classes and inherited CHM-v1 training mechanics:
- vocab 50,257
- d_model 512
- 24 layers
- 8 heads
- local window 512
- session length 1024 = two 512-token chunks
- LOCAL vs EIEM-FLAT only
- BF16 CUDA autocast
- AdamW, fused on CUDA when available
- betas `(0.9, 0.95)`
- weight decay `0.1`
- peak LR `3e-4`
- grad clip `1.0`
- micro-batch `4`
- grad accumulation `4`
- tokens/optimizer-step `16,384`
- no torch.compile
- exact frozen corpus `/vol/data/tam100m-2b-curated-v1`

This is not a scientific training run. Loss values may be checked only for finite/numerical sanity and must not be interpreted as modeling evidence.

## Frozen measured workload

For each model independently and sequentially, LOCAL first then EIEM-FLAT:
1. instantiate with engineering seed `977201` and verify exact Stage-A parameter counts/backbone initialization;
2. create the real AdamW optimizer;
3. execute **2 untimed warmup optimizer steps** using the real frozen corpus;
4. reset CUDA peak-memory stats and synchronize;
5. execute **8 measured optimizer steps** under the exact geometry above;
6. record measured tokens, wall seconds, tokens/s, peak allocated VRAM, finite-loss status, and final parameter finiteness.

Measured tokens per model are exactly `8 * 16,384 = 131,072`. Warmup observations are excluded from throughput projection.

The two models use deterministic but non-overlapping engineering sample streams derived from the same Stage-B seed. This is systems work, not a paired scientific comparison.

## Projection

For each model:
`projected_seconds = 33,554,432 / measured_tokens_per_second`

Also record:
- projected pair training seconds = LOCAL projected + EIEM projected
- EIEM/LOCAL throughput ratio
- projected compute cost using the live resource rate observed by the trigger workflow

Projection is descriptive systems evidence only; Stage C still requires a separate run-control and stronger deterministic evaluation suite.

## Predeclared PASS / STOP gate

Any of the following is an immediate `CHM_V1_100M_STAGE_B_SYSTEMS_STOP`:
- CUDA unavailable, OOM, exception, NaN/Inf loss, or non-finite parameters;
- exact parameter counts differ from Stage A;
- peak allocated VRAM exceeds `22.0 GiB` for either model;
- LOCAL measured throughput < `3,500 tokens/s`;
- EIEM-FLAT measured throughput < `2,500 tokens/s`;
- EIEM/LOCAL measured throughput ratio < `0.60`;
- projected pair training wall time for the frozen 33,554,432-token/model screen exceeds `21,600 s` (6 hours total);
- projected live-rate pair compute cost exceeds `$6.00`.

If every gate passes, classification is `CHM_V1_100M_STAGE_B_SYSTEMS_PASS`.

The thresholds are deliberately feasibility gates, not performance claims. Passing means only that one separately authorized Stage-C screen is operationally reasonable.

## Implementation / execution sequence

1. Add a dedicated Stage-B systems harness and unit guards on a fresh branch.
2. Add a guarded one-shot Modal launcher/workflow on that branch; merging it does **not** itself allocate GPU.
3. Require exact-PR-head CI PASS.
4. Merge only through PR; never write `main` directly.
5. Require exact-current-main CI PASS.
6. Record final launcher authority (source SHA/tree/harness blob/workflow blob and CI URL) on this issue.
7. Create the unique trigger issue exactly once.
8. Run zero-GPU preflight / corpus fingerprint / live-rate check first.
9. Persist a dispatch marker before allocation.
10. Execute the one L4 Stage-B attempt exactly once. No retry, rerun, rescue run, or redispatch.
11. Record the durable result and close this issue completed.

## Hard guards

- no execution of scientific seed `977001`
- no full 33,554,432-token/model training
- no scientific NLL/accuracy conclusion from Stage B
- no use of clipped indexed retrieval; Stage B models the future LOCAL vs EIEM-FLAT training path only
- no changes to CORTEX/AERA/RLT or unrelated research
- infrastructure failure is not scientific evidence, but the engineering seed/attempt remains consumed once GPU allocation starts
- PASS does not automatically authorize Stage C
- no novelty/SOTA/AGI/RSI/breakthrough claim

## Interpretation ceiling

Stage B can establish only whether the exact ~101.8M LOCAL/EIEM-FLAT pair is practically trainable on the frozen L4 envelope with sufficient VRAM/throughput headroom to justify preparing Stage C.

**Comment by vinceackermann2-sys** [comment]
CHM_V1_983_FINAL_LAUNCHER_AUTHORITY_V1
classification=FINAL_LAUNCHER_AUTHORITY_SYSTEMS_ONLY
source_sha=3ed002c0e0575792c6cea44c41b96311953241a6
source_tree=bee94f39eab7994c907b364b81d41adb2f4ebc02
harness_blob=518b3ee01f4acab70560eb7974862c360baa0f4f
workflow_blob=8f0b9eaf40cea17774299b8f55ef56e72a1e3506
ci_status=PASS
ci_url=https://github.com/vinceackermann2-sys/tam-research/actions/runs/34959423561
engineering_seed=977201
gpu=1x NVIDIA L4
cpu=4 physical cores
memory=16 GiB
max_gpu_seconds=1200
max_stage_b_compute_usd=0.50
scientific_seed_977001_authorized=false
automatic_retry=false
stage_c_authorized_automatically=false

This authority is limited to the one preregistered #983 non-scientific systems attempt. It does not authorize the 33,554,432-token/model scientific screen.
